multicloud/src/main/java/de/uniulm/omi/cloudiator/sword/multicloud/pricing/aws/AWSPricingSupplier.java

```diff
@@ -0,0 +1,204 @@
+                        break;
+                    }
+                    case "SUSE": {
+                        operatingSystemFamily = OperatingSystemFamily.SUSE;
+                        break;
+                    }
+                    case "Windows": {
+                        operatingSystemFamily = OperatingSystemFamily.WINDOWS;
+                        break;
+                    }
+                    default: {
+                        operatingSystemFamily = OperatingSystemFamily.UNKNOWN;
+                        break;
+                    }
+                }
+
+                OperatingSystemBuilder operatingSystemBuilder = OperatingSystemBuilder.newBuilder()
+                        .architecture(operatingSystemArchitecture)
+                        .family(operatingSystemFamily)
+                        .version(OperatingSystemVersions.unknown()); // AWS Pricing API exposes no information about version
+
+                List<PricingTerms> pricingOnDemandTerms = new ArrayList<>();
+                List<PricingTerms> pricingReservedTerms = new ArrayList<>();
+                Optional<String> currency = Optional.empty();
+
+                if(awsProductAndTerms.terms != null && awsProductAndTerms.terms.get("OnDemand") != null) {
+                    for (Map.Entry<String, AWSPriceDimensionsRates> entry : awsProductAndTerms.terms.get("OnDemand").entrySet()) {
+                        for (Map.Entry<String, AWSDimensionRates> entry2 : entry.getValue().getAwsDimensionRates().entrySet()) {
+                            List<PricingDimensions> pricingDimensions = new ArrayList<>();
+                            for (Map.Entry<String, AWSBaseTerm> entry3 : entry2.getValue().entrySet()) {
+                                Map.Entry<String, BigDecimal> price = entry3.getValue().getPricePerUnit().entrySet().iterator().next();
+                                pricingDimensions.add(new PricingDimensions(price.getValue(), entry3.getValue().getUnit(), entry3.getValue().getDescription(), entry3.getValue().getBeginRange(), entry3.getValue().getEndRange()));
+                            }
+                            pricingOnDemandTerms.add(new PricingTerms(pricingDimensions, entry.getValue().termAttributes.get("LeaseContractLength"), entry.getValue().termAttributes.get("OfferingClass"), entry.getValue().termAttributes.get("PurchaseOption")));
+                        }
+                    }
+                    currency = Optional.ofNullable(awsProductAndTerms.terms.get("OnDemand").entrySet().iterator().next().getValue().getAwsDimensionRates().entrySet().iterator().next().getValue().entrySet().iterator().next().getValue().getPricePerUnit().entrySet().iterator().next().getKey());
+                }
+
+                // currently only OnDemand offers are being presented
+                /*if (awsProductAndTerms.terms != null && awsProductAndTerms.terms.get("Reserved") != null) {
```

Comment:
Comment to remove?

Reply:
I would prefer to leave it, as Reserved instances are planned to be included in near future